### PR TITLE
add tfar 0.9.12 compatibility

### PR DESCRIPTION
**Pull request by mikehudak**

### CODE REVIEW

File: addons/radar/XEH_preInit.sqf [hunk 1]
```diff
@@ -27,6 +27,7 @@ GVAR(oddDirectionCompasses) setVariable ["lib_ger_itemcompass", 6400];
 GVAR(maxDegrees) = 360;
 
 private _tfar = isClass (configFile >> "CfgPatches" >> "tfar_core");
+private _tfarOld = isClass (configFile >> "CfgPatches" >> "task_force_radio");
```
Comment (diwako) [inline addons/radar/XEH_preInit.sqf:30]: Would say combine it with the other `_tfar` variable, no need to have two really if the events are the same

Suggestion:
```sqf
private _tfar = isClass (configFile >> "CfgPatches" >> "tfar_core") || {isClass (configFile >> "CfgPatches" >> "task_force_radio")};
```

File: addons/radar/XEH_postInit.sqf [hunk 2]
```diff
@@ -55,7 +55,9 @@ if (isClass(configfile >> "CfgPatches" >> "ace_finger")) then {
     }]  call CBA_fnc_addEventHandler;
 };
 
-if (isClass (configFile >> "CfgPatches" >> "tfar_core")) then {
+private _tfar = isClass (configFile >> "CfgPatches" >> "tfar_core");
```
Comment (diwako) [inline addons/radar/XEH_postInit.sqf:58]: Var declaration not really needed as they are not used afterwards. Combining both conditions should be still readable enough.

Suggestion:
```sqf
if (isClass (configFile >> "CfgPatches" >> "tfar_core") || {isClass (configFile >> "CfgPatches" >> "task_force_radio")}) then {
```

File: addons/radar/XEH_preInit.sqf (hunk 1 above)
Comment (dedmen) [reply addons/radar/XEH_preInit.sqf:30]: new TFAR also defines task_force_radio class, you can't use it to detect if you're running old version.
tfar_core missing means old

File: addons/main/XEH_preInit.sqf [hunk 3]
```diff
@@ -211,6 +211,7 @@ _curCat = localize "STR_dui_cat_custom_color";
 if !(hasInterface) exitWith {};
 
 GVAR(radioModSpectator) = configFile call {
+    if (isClass (_this >> "CfgPatches" >> "task_force_radio")) exitWith { {_player getVariable ["tf_forceSpectator", false]} };
```
Comment (dedmen) [inline addons/main/XEH_preInit.sqf:214]: This is wrong, this will return tf_forceSpectator even for TFAR beta which uses TFAR_forceSpectator

Comment (mikehudak) [reply addons/main/XEH_preInit.sqf:214]: so I should move it under TFAR beta check

File: addons/radar/XEH_preInit.sqf (hunk 1 above)
Comment (mikehudak) [reply addons/radar/XEH_preInit.sqf:30]: @dedmen can I use ?
`getText (configfile >> "CfgPatches" >> "task_force_radio" >> "version") == "0.9.12"`

**Comment by diwako** [comment]
Seeing that the cfgPatches class `task_force_radio` is present in both and `tfar_core` only in the beta this now could work. The only unsure thing are the CBA events. That certainly needs testing.

File: addons/radar/XEH_preInit.sqf (hunk 1 above)
Comment (Grezvany13) [reply addons/radar/XEH_preInit.sqf]: This works:
```
private _tfar = isClass (configFile >> "CfgPatches" >> "tfar_core");
private _tfarOld = isClass (configFile >> "CfgPatches" >> "task_force_radio") && !_tfar;
```
Returns correct values depending on loaded version

Comment (dedmen) [reply addons/radar/XEH_preInit.sqf]: no you cannot check version

**Comment by dedmen** [comment]
I think the events didnt change but its been too long

**Comment by diwako** [comment]
> I think the events didnt change but its been too long

I have taken a brief look into the current tfar master branch. From there I think it will work, as the objNull passed will convert it basically just to an CBA event. So far this has only been tested on TFAR 0.9.16.

I need some guinea pigs to test on current tfar version.

**Pull request merged**